pkg/api-server/dataplane_layout_endpoint.go

```diff
@@ -0,0 +1,186 @@
+		return
+	}
+
+	if mesh.Spec.GetMeshServices().GetMode() == v1alpha1.Mesh_MeshServices_Disabled {
+		response.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	dataplane, err := dle.getDataplaneResource(request.Request.Context(), meshName, dataplaneName)
+	if err != nil {
+		rest_errors.HandleError(request.Request.Context(), response, err, "Failed to retrieve Dataplane")
+		return
+	}
+
+	baseMeshContext, err := dle.meshContextBuilder.BuildBaseMeshContextIfChanged(request.Request.Context(), meshName, nil)
+	if err != nil {
+		rest_errors.HandleError(request.Request.Context(), response, err, "Failed to build MeshContext")
+	}
+
+	inbounds := util_slices.Map(dataplane.Spec.GetNetworking().GetInbound(), func(inbound *v1alpha1.Dataplane_Networking_Inbound) api_common.DataplaneInbound {
+		sectionName := inbound.GetName()
+		if sectionName == "" {
+			sectionName = strconv.Itoa(int(inbound.GetPort()))
+		}
+
+		return api_common.DataplaneInbound{
+			Kri:      kri.From(dataplane, sectionName).String(),
+			Port:     int32(inbound.GetPort()),
+			Protocol: inbound.GetProtocol(),
+		}
+	})
+
+	var outbounds []api_common.DataplaneOutbound
+	if dataplane.Spec.GetNetworking().GetOutbound() != nil {
+		// TODO handle user defined outbounds on universal without transparent proxy
+	} else {
+		// handle reachable backend and outbounds when using transparent proxy
+		reachableBackends := baseMeshContext.DestinationIndex.GetReachableBackends(mesh, dataplane)
+		outbounds = dle.getOutbounds(baseMeshContext.DestinationIndex, reachableBackends)
+	}
+
+	slices.SortStableFunc(outbounds, func(a, b api_common.DataplaneOutbound) int {
+		return strings.Compare(a.Kri, b.Kri)
+	})
+
+	networkingLayout := types.DataplaneNetworkingLayout{
+		Inbounds:  inbounds,
+		Kri:       kri.From(dataplane, "").String(),
+		Labels:    dataplane.GetMeta().GetLabels(),
+		Outbounds: outbounds,
+	}
+
+	err = response.WriteHeaderAndJson(http.StatusOK, networkingLayout, "application/json")
+	if err != nil {
+		log.Error(err, "Could not write response")
+	}
+}
+
+func (dle *dataplaneLayoutEndpoint) getMeshResource(ctx context.Context, meshName string) (*core_mesh.MeshResource, error) {
+	meshResource := core_mesh.NewMeshResource()
+	err := dle.resManager.Get(ctx, meshResource, store.GetByKey(meshName, core_model.NoMesh))
+	if err != nil {
+		return nil, err
+	}
+	return meshResource, nil
+}
+
+func (dle *dataplaneLayoutEndpoint) getDataplaneResource(ctx context.Context, meshName string, dataplaneName string) (*core_mesh.DataplaneResource, error) {
+	dataplaneResource := core_mesh.NewDataplaneResource()
+	err := dle.resManager.Get(ctx, dataplaneResource, store.GetByKey(dataplaneName, meshName))
+	if err != nil {
+		return nil, err
+	}
+	return dataplaneResource, nil
+}
+
+func (dle *dataplaneLayoutEndpoint) getOutbounds(destinationContext *xds_context.DestinationIndex, reachableBackends *xds_context.ReachableBackends) []api_common.DataplaneOutbound {
```

Comment:
looks like this could be a function and not need `dle`

Reply:
removed whole function